expr/func_countseries.go

```diff
@@ -0,0 +1,60 @@
+package expr
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/grafana/metrictank/api/models"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+type FuncCountSeries struct {
+	in []GraphiteFunc
+}
+
+func NewCountSeries() GraphiteFunc {
+	return &FuncCountSeries{}
+}
+
+func (s *FuncCountSeries) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesLists{val: &s.in},
+	}, []Arg{ArgSeries{}}
+}
+
+func (s *FuncCountSeries) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncCountSeries) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, queryPatts, err := consumeFuncs(cache, s.in)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(series) == 0 {
+		return series, nil
+	}
```

Comment:
interesting. would have thought we'd just return a series with vals=0 here, but i confirmed graphite does what you do.